src/htm/algorithms/SpatialPooler.cpp

```diff
@@ -363,7 +363,7 @@ void SpatialPooler::getConnectedSynapses(UInt column,
 void SpatialPooler::getConnectedCounts(UInt connectedCounts[]) const {
   for(UInt seg = 0; seg < numColumns_; seg++) {
     const auto &segment = connections_.dataForSegment( seg );
-    connectedCounts[ seg ] = segment.numConnected;
+    connectedCounts[ seg ] = segment.numConnected; //TODO numConnected only used here, rm from SegmentData and compute for each segment.synapses?
```

Comment:
plan to test if this should be moved from SegData, and computed only as a lambda here. 
This is the only place where this is used, but updated each updateSynapsePermanence

Reply:
I'd like to keep `connectedCounts` computed and on hand
- The synapse competition uses it
- The (Kropff & Treves 2008) model uses it
- It's fast to compute, its a small amount of book keeping that gets done when a synapse changes between connected/disconnected

Maybe these applications can be optimized to not need this data, or maybe there is a different way to optimize this, but for now I'd rather not remove this.

Reply:
> The synapse competition uses it
The (Kropff & Treves 2008) model uses it

sure, let's keep it if you have coming users for that!